docs/self-hosted/kubernetes-migration-guide.md

```diff
@@ -0,0 +1,144 @@
+a Redis database used as secondary storage, cache, and job queue, and a file/blob storage for uploaded artifacts and dist
+files used during composer install.
+
+To avoid any data inconsistency and random errors during application usage all three of them need to be backed up and restored at the same time.
+
+Please note that you won't be able to restore Private Packagist Self-Hosted Kubernetes from snapshots created via Private Packagist Self-Hosted Replicated Native.
+
+### Stop the Private Packagist application
+
+```
+docker stop packagist-worker packagist-web packagist-repo
+```
+
+### Backup the dist and artifact files
+
+At the end of this step, you should have a `packagist_storage.tar.gz` file in your working directory.
+
+Please note that depending on the number of packages and versions used in your projects and the size of your dependencies
+it can take several minutes for this command to finish.
+
+```
+docker exec -it packagist-ui /srv/manager/bin/console packagist:self-hosted:migrate-storage export
+docker cp packagist-ui:/tmp/storage.tar.gz packagist_storage.tar.gz
+docker exec -it packagist-ui rm /tmp/storage.tar.gz
+docker stop packagist-ui
+```
+
+### Backup the PostgreSQL database
+
+At the end of this step, you should have a `packagist_db.sql` file in your working directory.
+
+```
+docker exec -it packagist-postgres pg_dump --clean --if-exists -U postgres -d packagist_db > packagist_db.sql
+docker stop packagist-postgres
+```
+
+### Backup the Redis database
+
+At the end of this step, you should have a `packagist_redis.rdb` file in your working directory.
+
+```
+docker exec -it packagist-redis /usr/local/bin/redis-backup.sh
+docker cp packagist-redis:/data/dump.rdb packagist_redis.rdb
+docker stop packagist-redis
+```
+
+### Start Private Packagist again (optional)
+
+Now that you have exported all the data, you may restart the Private Packagist application. This will allow your developers
+to continue with their work while you import the data in the Private Packagist Self-Hosted Kubernetes setup.
+
+```
+docker start packagist-postgres packagist-redis packagist-ui packagist-repo packagist-web packagist-worker    
+```
+
+Please note that any changes made in Private Packagist will now be lost and will have to be reapplied once everyone switches to Private Packagist Self-Hosted Kubernetes.
+
+## Installing Private Packagist Self-Hosted Kubernetes
+If you haven't done so already, now is the time to follow the [installation guide](./kubernetes.md) to install Private Packagist Self-Hosted Kubernetes.
+
+## Restore application data in Kubernetes
+
+All commands assume that you are logged in to the server where Private Packagist Self-Hosted Kubernetes is installed,
+your working directory contains the three files generated by the previous steps, and that
+Private Packagist Self-Hosted Kubernetes is installed in the Kubernetes namespace "default".
```

Comment:
We should recommend to use the default namespace or not?

Maybe not add -n default to all commands but switch the namespace globally? Dont know, just thinking out loud

Reply:
+1 on removing the namespace. Makes them easier to port with someone uses a different namespace. I would assume anyone not using the embedded cluster won't use the default namespace. Definitely wouldn't recommend that